src/VisualStudio/Core/Def/Implementation/ProjectSystem/RunningDocumentTableEventTracker.cs

```diff
@@ -0,0 +1,239 @@
+            {
+                if (CheckPreconditions(docCookie))
+                {
+                    var args = new RunningDocumentTableEventArgs(docCookie, _runningDocumentTable.GetDocumentMoniker(docCookie));
+                    OnReloadDocumentData?.Invoke(this, args);
+                }
+            }
+
+            if ((grfAttribs & (uint)__VSRDTATTRIB.RDTA_Hierarchy) != 0)
+            {
+                var args = new RunningDocumentTableEventArgs(docCookie, _runningDocumentTable.GetDocumentMoniker(docCookie));
+                OnRefreshDocumentContext?.Invoke(this, args);
+            }
+
+            return VSConstants.S_OK;
+        }
+
+        public int OnBeforeDocumentWindowShow(uint docCookie, int fFirstShow, IVsWindowFrame pFrame)
+        {
+            if (fFirstShow != 0)
+            {
+                if (TryGetBuffer(docCookie, out var buffer))
+                {
+                    var args = new RunningDocumentTableInitializedEventArgs(docCookie, _runningDocumentTable.GetDocumentMoniker(docCookie), buffer);
+                    OnBeforeOpenDocument?.Invoke(this, args);
+                }
+            }
+
+            return VSConstants.S_OK;
+        }
+
+        public int OnAfterDocumentWindowHide(uint docCookie, IVsWindowFrame pFrame)
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+
+        public int OnBeforeSave(uint docCookie)
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+
+        /// <summary>
+        /// Gets the text buffer for a document cookie.
+        /// Also checks to make sure the document is initialized before returning.
+        /// </summary>
+        public bool TryGetBuffer(uint docCookie, out ITextBuffer textBuffer)
+        {
+            textBuffer = null;
+            if (!CheckPreconditions(docCookie))
+            {
+                return false;
+            }
+
+            if ((object)_runningDocumentTable.GetDocumentData(docCookie) is IVsTextBuffer bufferAdapter)
+            {
+                textBuffer = _editorAdaptersFactoryService.GetDocumentBuffer(bufferAdapter);
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Checks that we're on the UI thread and that the document has already been initialized.
+        /// </summary>
+        private bool CheckPreconditions(uint docCookie)
+        {
+            _foregroundAffinitization.AssertIsForeground();
+
+            if (!_runningDocumentTable.IsDocumentInitialized(docCookie))
+            {
+                // We never want to touch documents that haven't been initialized yet, so immediately bail. Any further
+                // calls to the RDT might accidentally initialize it.
+                return false;
+            }
+
+            return true;
+        }
+
+        #region IDisposable Support
+        private bool _disposedValue = false; // To detect redundant calls
```

Comment:
>_disposedValue [](start = 21, length = 14)

I'd name this `_isDisposed` and move it to the top of the type declaration.

Reply:
Will do... perhaps we should update our code fix if that's the preferred name :)